src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -373,8 +327,6 @@ static int gethostbyaddr_r(const uint8_t* addr, const socklen_t len, int type, s
         return GetHostErrorCodes_BAD_ARG;
     }
 
-    pthread_mutex_lock(&lock_hostbyx_mutex);
```

Comment:
I think mutex locks need to stay.

Reply:
ok let me understand the reason, maybe "The functions gethostbyname() and gethostbyaddr() may return pointers to static data, which may be overwritten by later calls. Copying the struct hostent does not suffice, since it contains pointers; a deep copy is required."  [here](https://linux.die.net/man/3/gethostbyaddr), thank's for help, i'm not a "unix kernel expert".

Reply:
Yes, all three `pthread_mutext_t` changes can be undone.

See the comment in the removed code:

```diff
-/*
-Note: we're assuming that all access to these functions are going through these shims on the platforms, which do not provide
-      thread-safe functions to get host name or address. If that is not the case (which is very likely) race condition is
-      possible, for instance; if other libs (such as libcurl) call gethostby[name/addr] simultaneously.
-*/
-static pthread_mutex_t lock_hostbyx_mutex = PTHREAD_MUTEX_INITIALIZER;
-
```

The race-condition occurs when we call a shared lib that doesn't provide thread-safety, then we need to implement thread-safety by means of mutex lock on accessing the shared resource from that lib and hope that other consumer of that resource are doing the same (OS-wide).

Therefore, mutex is still relevant and unrelated to `SystemNative_GetHostByName` changes.

Reply:
Ok i agree i'll revert!